PackageExplorer/MainWindow.xaml.cs

```diff
@@ -146,33 +148,45 @@ private bool OpenLocalPackageCore(string packagePath)
         {
             IPackage package = null;
 
+            var tempFile = Path.GetTempFileName();
             try
             {
+                File.Copy(packagePath, tempFile, overwrite: true);
+
                 var extension = Path.GetExtension(packagePath);
                 if (extension.Equals(Constants.PackageExtension, StringComparison.OrdinalIgnoreCase))
                 {
-                    package = new ZipPackage(packagePath);
+                    package = new ZipPackage(tempFile);
                 }
                 else if (extension.Equals(Constants.ManifestExtension, StringComparison.OrdinalIgnoreCase))
                 {
-                    using (var str = ManifestUtility.ReadManifest(packagePath))
+                    using (var str = ManifestUtility.ReadManifest(tempFile))
                     {
-                        var builder = new PackageBuilder(str, Path.GetDirectoryName(packagePath));
+                        var builder = new PackageBuilder(str, Path.GetDirectoryName(tempFile));
                         package = builder.Build();
                     }
                 }
 
                 if (package != null)
                 {
                     LoadPackage(package, packagePath, PackageType.LocalPackage);
+                    _tempFile = tempFile;
                     return true;
                 }
             }
             catch (Exception ex)
             {
+                package = null;
                 UIServices.Show(ex.Message, MessageLevel.Error);
                 return false;
             }
+            finally
+            {
+                if (package == null && File.Exists(tempFile))
+                {
+                    File.Delete(tempFile);
```

Comment:
This could still fail as File.Exists is a race condition, and File.Delete could throw. We should try/catch it and swallow.